lua/leetcode/api/question.lua

```diff
@@ -37,25 +37,24 @@ function question.random()
     local query = queries.random_question
 
     local config = require("leetcode.config")
-    local res, err = utils.query(query, variables)
-    if err then return nil, err end
-
-    local q = res.data.randomQuestion
-    if config.is_cn then
-        q = {
-            title_slug = q,
-            paid_only = problemlist.get_by_title_slug(q).paid_only,
-        }
-    end
-
-    if not config.auth.is_premium and q.paid_only then
-        err = err or {}
-        err.msg = "Drawn question is for premium users only. Please try again"
-        err.lvl = vim.log.levels.WARN
-        return nil, err
-    end
-
-    return q
+    repeat
+        local res, err = utils.query(query, variables)
+        if err then return nil, err end
+
+        local q = res.data.randomQuestion
+        if config.is_cn then
+            q = {
+                title_slug = q,
+                paid_only = problemlist.get_by_title_slug(q).paid_only,
+            }
+        end
+
+        if not config.auth.is_premium and q.paid_only then
+            vim.notify("Drawn question is for premium users only. Reattempting to get another question.")
+        else
+            return q
+        end
+    until false
```

Comment:
Thank you for your contribution.

I've thought of this, but if someone has a slow internet and it re-fetches the random question like 5 times, it might just lead to some unpleasant experience since it will completely block the thread for the duration of that time.

I think the way to go is to just make it optional.
So you could do something like that

```lua
local res, err = utils.query(query, variables)
if err then return nil, err end

local q = res.data.randomQuestion
if config.is_cn then
    q = {
        title_slug = q,
        paid_only = problemlist.get_by_title_slug(q).paid_only,
    }
end

if not config.auth.is_premium and q.paid_only then
    if not config.user.auto_retry_random_question then
        err = err or {}
        err.msg = "Drawn question is for premium users only. Please try again"
        err.lvl = vim.log.levels.WARN
        return nil, err
    else
        question.random()
else
    return q
end
```
i have not checked if it works tho